Tpetra: configure error using COMPLEX_DOUBLE and FLOAT types

<!---
Replace <teamName> below with the appropriate Trilinos package/team name.
-->
@trilinos/Tpetra




## Expectations

We do not want to turn on Trilinos_ENABLE_COMPLEX_FLOAT if we are also using Trilinos_ENABLE_FLOAT and Trilinos_ENABLE_COMPLEX_DOUBLE.  Is the current behavior (below) correct? 

<!---
Tell us what you think should happen, how you think things should work, what
you would like to see in the documentation, etc.



-->

## Current Behavior
Processing enabled package: Tpetra (Classic, TSQR, Core)
-- Tpetra: Enabling deprecated code
-- Determine whether Tpetra will assume that MPI is CUDA aware:
--   - Tpetra_ENABLE_CUDA is OFF, so Tpetra will assume that MPI is not CUDA aware.
-- Tpetra execution space availability (ON means available): 
--   - Serial:  ON
--   - Threads: OFF
--   - OpenMP:  OFF
--   - Cuda:    OFF
CMake Error at packages/tpetra/CMakeLists.txt:986 (MESSAGE):
  Tpetra: Tpetra_INST_COMPLEX_FLOAT is OFF (meaning that you want to disable
  explicit instantiation and/or tests of Tpetra with Scalar =
  std::complex<float>), but Teuchos_ENABLE_COMPLEX is ON and
  Teuchos_ENABLE_FLOAT is ON.  This will cause downstream link errors, likely
  in Thyra or Stratimikos.  See GitHub Issue 4080 for details.  Best
  practice: Do NOT set Tpetra_INST_COMPLEX_FLOAT, Teuchos_ENABLE_COMPLEX, or
  Teuchos_ENABLE_FLOAT explicitly.  Instead, set
  Trilinos_ENABLE_COMPLEX_FLOAT explicitly.  That will set defaults for
  Teuchos and Tpetra correctly.


-- Configuring incomplete, errors occurred!
See also "/scratch/ajpowel/code_032119/fix_trilly/CMakeFiles/CMakeOutput.log".
See also "/scratch/ajpowel/code_032119/fix_trilly/CMakeFiles/CMakeError.log".



<!---
Tell us how the current behavior fails to meet your expectations in some way.
-->

## Motivation and Context

Sierra-Trilinos integration 


<!---
How has this expectation failure affected you?  What are you trying to
accomplish?  Why do we need to address this?  What does it have to do with
anything?  Providing context helps us come up with a solution that is most
useful in the real world.
-->


## Steps to Reproduce

cmake -DTrilinos_ENABLE_Tpetra=ON -DTrilinos_ENABLE_COMPLEX_DOUBLE=ON -DTrilinos_ENABLE_FLOAT=ON -DTrilinos_ENABLE_Teuchos=ON $PATH_TO_MY_SOURCE_CODE

<!---
Provide a link to a live example, or an unambiguous set of steps to reproduce
this issue.  Include code to reproduce, if relevant. 
1. Do this.
1. Do that.
1. Shake fist angrily at computer.
-->

--- a/packages/tpetra/CMakeLists.txt
+++ b/packages/tpetra/CMakeLists.txt
@@ -876,12 +876,6 @@ IF (Tpetra_INST_FLOAT)
     MESSAGE(FATAL_ERROR "Tpetra: Tpetra_INST_FLOAT is ON (meaning that you want to instantiate and/or test Tpetra with Scalar = float), but HAVE_TEUCHOS_BLASFLOAT is OFF.  This means that you are linking with a BLAS library that lacks float (S) support.  Tpetra needs a BLAS implementation that supports float.")
   ENDIF ()
 
-ELSE () # Tpetra_INST_FLOAT is OFF
-
-  IF (DEFINED Teuchos_ENABLE_FLOAT AND Teuchos_ENABLE_FLOAT)
-    MESSAGE(FATAL_ERROR "Tpetra: Tpetra_INST_FLOAT is OFF (meaning that you want to disable explicit instantiation and/or tests of Tpetra with Scalar = float), but Teuchos_ENABLE_FLOAT is ON.  This will cause downstream link errors, likely in Stratimikos.  See GitHub Issues 4059 and 4080 for details.  Best practice: Do NOT set Tpetra_INST_FLOAT OR Teuchos_ENABLE_FLOAT explicitly.  Instead, set Trilinos_ENABLE_FLOAT explicitly.  That will set defaults for Teuchos and Tpetra correctly.")
-  ENDIF ()
-
 ENDIF ()
 
 # Decide whether to instantiate / test Scalar = std::complex<float> and/or
@@ -1000,14 +994,6 @@ IF (Tpetra_INST_COMPLEX_FLOAT)
     MESSAGE(FATAL_ERROR "Tpetra: Tpetra_INST_COMPLEX_FLOAT is ON (meaning that you want to want to instantiate and/or test Tpetra classes with Scalar = std::complex<float>), but HAVE_COMPLEX_BLAS is not defined.  This means that you are linking with a BLAS library that lacks complex arithmetic (Z) support.  Tpetra needs a BLAS implementation that supports complex arithmetic.")
   ELSEIF (NOT HAVE_COMPLEX_BLAS)
     MESSAGE(FATAL_ERROR "Tpetra: Tpetra_INST_COMPLEX_FLOAT is ON (meaning that you want to want to instantiate and/or test Tpetra classes with Scalar = std::complex<float>), but HAVE_COMPLEX_BLAS is OFF.  This means that you are linking with a BLAS library that lacks complex arithmetic (Z) support.  Tpetra needs a BLAS implementation that supports complex arithmetic.")
-  ENDIF ()
-
-ELSE () # Tpetra_INST_COMPLEX_FLOAT is OFF
-
-  IF (DEFINED Teuchos_ENABLE_COMPLEX AND Teuchos_ENABLE_COMPLEX)
-    IF (DEFINED Teuchos_ENABLE_FLOAT AND Teuchos_ENABLE_FLOAT)
-      MESSAGE(FATAL_ERROR "Tpetra: Tpetra_INST_COMPLEX_FLOAT is OFF (meaning that you want to disable explicit instantiation and/or tests of Tpetra with Scalar = std::complex<float>), but Teuchos_ENABLE_COMPLEX is ON and Teuchos_ENABLE_FLOAT is ON.  This will cause downstream link errors, likely in Thyra or Stratimikos.  See GitHub Issue 4080 for details.  Best practice: Do NOT set Tpetra_INST_COMPLEX_FLOAT, Teuchos_ENABLE_COMPLEX, or Teuchos_ENABLE_FLOAT explicitly.  Instead, set Trilinos_ENABLE_COMPLEX_FLOAT explicitly.  That will set defaults for Teuchos and Tpetra correctly.")
-    ENDIF ()
   ENDIF ()
 
 ENDIF ()

